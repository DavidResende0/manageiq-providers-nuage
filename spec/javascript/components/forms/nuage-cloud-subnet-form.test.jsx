import NuageCloudSubnetForm from '../../../../app/javascript/components/forms/nuage-cloud-subnet-form'

let renderComponent;

describe('NuageCloudSubnetForm', () => {
  beforeAll(() => {
    renderComponent = (loading = false) => shallowRedux(<NuageCloudSubnetForm updateFormState={jest.fn()} loading={loading} />);
  });

  describe('renders', () => {
    it('form', () => {
      let component = renderComponent();
      expect(toJson(component)).toMatchSnapshot();
    });

    it('spinner', () => {
      let component = renderComponent(true);
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
