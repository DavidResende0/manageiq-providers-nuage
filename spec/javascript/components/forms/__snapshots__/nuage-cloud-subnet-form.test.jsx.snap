// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NuageCloudSubnetForm renders form 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          label="_Name_"
          name="name"
          parse={[Function]}
          placeholder="Subnet Name"
          validate={[Function]}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          label="_Address_"
          name="address"
          parse={[Function]}
          placeholder="100.100.100.0"
          validate={[Function]}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          label="_Netmask_"
          name="netmask"
          parse={[Function]}
          placeholder="255.255.255.0"
          validate={[Function]}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          label="_Gateway_"
          name="gateway"
          parse={[Function]}
          placeholder="100.100.100.1"
          validate={[Function]}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`NuageCloudSubnetForm renders spinner 1`] = `
<div
  className="spinner spinner-lg"
/>
`;
